### PR TITLE
Add test for pip installations

### PR DESCRIPTION
**Pull request by lxn2**
This commit adds Dockerfiles and test script to test all variants of MXNet pip packages. I've tested this on a copy of the Jenkins slave host with an updated NVIDIA driver (375.26) with the following command:

`tests/jenkins/run_test_pip_installations.sh ${WORKSPACE}` where WORKSPACE was set to /home/ec2-user/mxnet, the Jenkins workspace.

Currently the test will fail after testing `mxnet` with Python 3.6 as the accuracy is lower than expected. We can either put that case at the very end or change the script to not fail until all cases have been tested.

@sandeep-krishnamurthy

### CODE REVIEW

File: tests/ci_build/pip_tests/Dockerfile.pip_dependencies [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# -*- mode: dockerfile -*-
+# part of the dockerfile to test pip installations
+
+# add repo to install different Python versions
+RUN apt-get update && apt-get install -y software-properties-common
+RUN add-apt-repository ppa:fkrull/deadsnakes && apt-get update
```
Comment (sandeep-krishnamurthy) [inline tests/ci_build/pip_tests/Dockerfile.pip_dependencies]: Why is this required?

File: tests/jenkins/run_test_pip_installations.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    fi
+
+    # concatenate the Dockerfile with dependencies into the device file
+    DOCKERFILE="Dockerfile.${DEV}"
+    DOCKERFILE_DEVICE="Dockerfile.in.${DEV}"
+    rm -rf ${DOCKERFILE}
+    cp ${DOCKERFILE_DEVICE} ${DOCKERFILE}
+    cat Dockerfile.pip_dependencies >> ${DOCKERFILE}
+
+    # build Docker image
```
Comment (sandeep-krishnamurthy) [inline tests/jenkins/run_test_pip_installations.sh]: Why do we want to build docker image every time?

File: tests/ci_build/pip_tests/Dockerfile.pip_dependencies (hunk 1 above)
Comment (lxn2) [inline tests/ci_build/pip_tests/Dockerfile.pip_dependencies]: This repository has a lot of old/new Python versions to test with. This allows me to install multiple versions of Python 3.

File: tests/jenkins/run_test_pip_installations.sh (hunk 2 above)
Comment (lxn2) [inline tests/jenkins/run_test_pip_installations.sh]: 1. It can rebuild if we've made changes to the Dockerfile
2. We don't assume that the image exists on the host
3. We don't have to rely on Dockerhub/ECS and push small images that we only use for testing purposes

What's nice about this is that if we've rerun this test in succession, the image built the first time is cached, so we don't spend additional time rebuilding.

**Comment by lxn2** [comment]
nosetest failing possibly because I added arguments to the test_conv.py script. Debugging and will push the changes after.

**Comment by mli** [comment]
@szha should we put the pip tests into the pip build repo?

**Comment by szha** [comment]
It would be nice to include tests after the build of each variation. The blocker was the Travis build time-limit. We will need to move the release to Jenkins if we want tests like this pr. Currently I manually test all the releases with unit tests.

**Comment by piiswrong** [comment]
if we have mac on Jenkins we can move the release to Jenkins

**Comment by lxn2** [comment]
@mli @szha @piiswrong we don't have mac on Jenkins right now and can't prioritize adding that. Can we push this for now so we can run a nightly test?

I can organize this test script under tests/jenkins/pip/ folder if that helps.
